web/ui/react-app/src/pages/alerts/AlertContents.tsx

```diff
@@ -100,6 +106,14 @@ const AlertsContent: FC<AlertsProps> = ({ groups = [], statsCount }) => {
     }
   };
 
+  const defaultValue = useMemo(() => {
+    const locationSearch = window.location.search;
+    const params = new URLSearchParams(locationSearch);
+    const search = params.get('expr') || '';
+    handleSearchChange(search);
```

Comment:
Causing side effects from a `useMemo` call is a bit unexpected (even though it might work in this case). See also the React docs: "side effects belong in useEffect, not useMemo." (https://reactjs.org/docs/hooks-reference.html#usememo)

I was thinking rather than calling `handleSearchChange()` here in every page component, why not just add a one-time `useEffect(() => handleChange(defaultValue), [])` directly in the search bar component that calls the search change handler once when the search bar initially mounts? Then there is also less code repetition.

Reply:
ready! :)